packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,448 @@
+import { PsbtV2 } from "@caravan/psbt";
+import { Network } from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { createOutputScript } from "./utils";
+import { DEFAULT_DUST_THRESHOLD, RBF_SEQUENCE } from "./constants";
+import { RbfOptions, FeeRate, UrgencyLevel } from "./types";
+
+/**
+ * RbfTransaction Class
+ *
+ * This class implements the Replace-By-Fee (RBF) functionality for Bitcoin transactions,
+ * adhering to BIP125 (https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki).
+ * It allows for the creation of replacement transactions with higher fees, either to
+ * accelerate confirmation or to cancel a previous transaction.
+ *
+ * The class supports both transaction acceleration and cancellation, with configurable
+ * fee rates based on network conditions and user-defined urgency levels.
+ *
+ * Key Features:
+ * - Supports PSBTv2 and PSBTv0 formats
+ * - Implements BIP125 rules for RBF
+ * - Allows fee subtraction from specific outputs or proportional distribution
+ * - Supports additional input selection for fee coverage
+ * - Provides transaction cancellation functionality
+ *
+ * Usage:
+ * const rbfTx = new RbfTransaction(psbt, feeMarket, options, network);
+ * const acceleratedPsbt = rbfTx.prepareAccelerated();
+ * // or
+ * const canceledPsbt = rbfTx.prepareCanceled(destinationAddress);
+ */
+export class RbfTransaction {
+  private originalPsbt: PsbtV2;
+  private newPsbt: PsbtV2;
+  private options: RbfOptions;
+  private feeMarket: FeeRate;
+  private urgencyMultipliers: Record<UrgencyLevel, number>;
+  private network: Network;
+
+  /**
+   * Constructor for RbfTransaction
+   *
+   * @param {PsbtV2 | string | Buffer} psbt - The original PSBT to be replaced
+   * @param {FeeRate} feeMarket - Current fee market rates
+   * @param {RbfOptions} options - Configuration options for the RBF transaction
+   * @param {Network} network - The Bitcoin network (mainnet, testnet, etc.)
+   *
+   * @throws {Error} If the PSBT is not in a valid format or not signaling RBF
+   */
+  constructor(
+    psbt: PsbtV2 | string | Buffer,
+    feeMarket: FeeRate,
+    options: RbfOptions,
+    network: Network,
+  ) {
+    this.originalPsbt = this.initializePsbt(psbt);
+    this.newPsbt = new PsbtV2(); // Create a new empty PsbtV2
+    this.originalPsbt.copy(this.newPsbt); // Copy the original PSBT to the new one
+    this.feeMarket = feeMarket;
+    this.network = network;
+    this.options = {
```

Comment:
My personal preference is that once you're in the class, we should need a nested options attribute. These can be just accessed directly. no need in my opinion to differentiate between this.network and this.options.dustThreshold. They're all just class attributes at this point. 

this also makes the typing a little clearer at the top of the class and lets you do things like make individual properties readonly, private, public.

Reply:
Addressed in commit [Refactored RBF and CPFP classes, introduced TransactionAnalyzer class](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a)